core/src/main/scala/sttp/tapir/EndpointIO.scala

```diff
@@ -421,7 +421,9 @@ object EndpointIO {
 
   //
 
-  case class Example[+T](value: T, name: Option[String], summary: Option[String])
+  case class Example[+T](value: T, name: Option[String], summary: Option[String]) {
+    def map[B](f: T => B): Example[B] = Example(f(value), name = name, summary = summary)
```

Comment:
Why not use `.copy` and only change `value`? It'll be more future-proof if more fields ever get added :)

Reply:
copy does not change the type param

Reply:
Are you sure?

```scala
Welcome to the Ammonite Repl 2.0.4 (Scala 2.13.1 Java 1.8.0_202)
@ case class Example[+T](value: T, name: Option[String], summary: Option[String]) {
    def map[B](f: T => B): Example[B] = Example(f(value), name = name, summary = summary)
  }
defined class Example
```

Reply:
Are you sure?

```scala
Welcome to the Ammonite Repl 2.0.4 (Scala 2.13.1 Java 1.8.0_202)
@ case class Example[+T](value: T, name: Option[String], summary: Option[String]) {
    def map[B](f: T => B): Example[B] = copy(value = f(value))
  }
defined class Example
```

Reply:
hmm apparently you're right, however I'd bet once I've originally implemented that, copy was complaining :D